core/trino-main/src/main/java/io/trino/operator/output/PartitionedOutputOperator.java

```diff
@@ -55,6 +56,7 @@
         private final boolean replicatesAnyRow;
         private final OptionalInt nullChannel;
         private final DataSize maxMemory;
+        private final PositionsAppenderFactory positionsAppenderFactory;
```

Comment:
maybe `BlockAppenderFactory appenderFactory` would be better (also for RLE use case later)